src/rdo.rs

```diff
@@ -629,7 +629,21 @@ fn luma_chroma_mode_rdo<T: Pixel>(
         x if x >= 4. => 2,
         _ => unreachable!(),
       };
-      heuristic_sidx..=heuristic_sidx
+      // prevent the highest sidx from bringing us into lossless
+      if fi.base_q_idx as i16
+        + ts.segmentation.data[heuristic_sidx as usize]
+          [SegLvl::SEG_LVL_ALT_Q as usize]
+        < 1
+      {
+        heuristic_sidx - 1..=heuristic_sidx - 1
```

Comment:
As this is routing around the current segementation design, I note that:
- We expect to only encounter this condition when `heuristic_sidx == 2` and maps to `-TEMPORAL_RDO_QI_DELTA`
- In that case, `heuristic_sidx - 1 == 1` and maps to `+TEMPORAL_RDO_QI_DELTA`
- The remaining choice of `heuristic_sidx - 2 == 0` maps to zero and is a closer match.

```suggestion
        0..=0
```